wagtailio/core/blocks.py

```diff
@@ -188,6 +188,13 @@ class Meta:
         template = "core/blocks/home_page_block.html"
 
 
+class CTALinkStructValue(blocks.StructValue):
```

Comment:
Praise: I really like this approach, I suggest to improve this slightly by refactoring existing implementations that handle this logic as well. (Search for usage of the CTABlock model) as we're handling this in context https://github.com/wagtail/wagtail.org/blob/513c9c14323b934cce1e046dc468f5191be5599d/wagtailio/core/blocks.py#L284. We could also add validation for supplying on or the other (or none). What do you think?

Reply:
Sounds great, thanks! TIL more of how to take advantage of mixins!

Updated in 3687292